### PR TITLE
[NOT FOR UPSTREAM] drm/xe/lnl: Drop force_probe requirement

### PR DESCRIPTION
**Pull request by kv2019i**
Enable the Xe driver on all test devices and enable gpu_bind in SOF.

### CODE REVIEW

**Comment by kv2019i** [comment]
This has some preconditions on the CI machines, so I'll mark as draft. Once @ssavati or @marc-hb acks CI is clear, please go ahaed with merge.

**Review by plbossart — APPROVED** [review]

**Comment by plbossart** [comment]
@ssavati @marc-hb and @fredoh9 can you ack that the CI devices have been updated?

**Comment by ssavati** [comment]
> @ssavati @marc-hb and @fredoh9 can you ack that the CI devices have been updated?

@plbossart I have checked all are updated

**Comment by kv2019i** [comment]
SOFCI TEST

**Comment by kv2019i** [comment]
Seems we are hitting a Xe driver load error on one system, so need more testing before this can be merged. https://gitlab.freedesktop.org/drm/xe/kernel/-/issues/1631 is similar (same error message), but not the same use-case.
UPDATE: referring to failure in https://sof-ci.01.org/linuxpr/PR5110/build4106/devicetest/index.html?model=LNLM_RVP_HDA&testcase=verify-kernel-boot-log

**Comment by kv2019i** [comment]
SOFCI TEST

**Comment by marc-hb** [comment]
We may have inconsistent GuC versions across devices - we still haven't productized Ansible :-(

**Comment by kv2019i** [comment]
The results look ok now. There's one fail on https://sof-ci.01.org/linuxpr/PR5110/build4324/devicetest/index.html?model=LNLM_RVP_HDA&testcase=check-sof-logger I can't explain (no dmesg), but rest of the tests are good so appears Xe works on that DUT and we have SOF probing.

So @ssavati and @marc-hb if you don't object, I'd proceed with merging this as handle any DUT specific issues as we find them.

**Comment by ssavati** [comment]
> So @ssavati and @marc-hb if you don't object, I'd proceed with merging this as handle any DUT specific issues as we find them.

@kv2019i I rerun it again "planresultdetail/44656" issue did not reproduced. from my side i think it safe merge. if any devices/test specifc issue we can track it later.

**Comment by marc-hb** [comment]
Fine by me since this does not add more red that would hide actual audio issues.

**Comment by marc-hb** [comment]
SOFCI TEST

(re-running because some devices were unavailable)

**Comment by marc-hb** [comment]
Very strange, this PR seems to break... `check-sof-logger` on `CML_HEL_RT5682-ipc3` !?! This configuration does not use the `xe` driver at all! Yet it failed twice back to back:

https://sof-ci.01.org/linuxpr/PR5110/build4326/devicetest/index.html?model=CML_HEL_RT5682-ipc3&testcase=check-sof-logger
https://sof-ci.01.org/linuxpr/PR5110/build4334/devicetest/index.html?model=CML_HEL_RT5682-ipc3&testcase=check-sof-logger

```
error: log_entry_address          0 is not in dictionary range!
```

Today's daily build 44658 did NOT fail and this test either rarely or never fails in that configuration.
Linux Commit: [bc47b82db6e0](https://github.com/thesofproject/linux/commit/bc47b82db6e03d540061964d4540a371e7d344c8)
KConfig Commit: [8189104a4f38](https://github.com/thesofproject/kconfig/commit/8189104a4f38167aaaa4e70eece20ba4b6c19ef1)
SOF Commit: [e05ca349de3d](https://github.com/thesofproject/sof/commit/e05ca349de3d65a093fbbe09a3390306755e9a30)

This recent PR did not fail either: https://sof-ci.01.org/linuxpr/PR5140/build4314/devicetest/index.html

**Comment by marc-hb** [comment]
SOFCI TEST

EDIT: https://sof-ci.01.org/linuxpr/PR5110/build4338/devicetest/index.html looks like it's passing now.

The sof-logger has never been 100% reliable, so maybe it was just incredibly bad luck that it failed twice back to back in the same configuration...?

**Comment by kv2019i** [comment]
@ujfalusi @bardliao @ranj063 feel free to proceed

File: drivers/gpu/drm/xe/xe_pci.c [hunk 1]
```diff
@@ -336,7 +336,6 @@ static const struct xe_device_desc mtl_desc = {
 static const struct xe_device_desc lnl_desc = {
 	PLATFORM(XE_LUNARLAKE),
 	.has_display = true,
-	.require_force_probe = true,
```
Comment (ujfalusi) [inline drivers/gpu/drm/xe/xe_pci.c:339]: Don't we have similar patch in linux-next to sherry-pick?

Comment (kv2019i) [reply drivers/gpu/drm/xe/xe_pci.c:339]: Let me check, when I submitted this (one month ago), there wasn't anything in linx-next yet. Btw, sherry-picking sounds fun :)

Comment (kv2019i) [reply drivers/gpu/drm/xe/xe_pci.c:339]: @ujfalusi nope, not yet https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/drivers/gpu/drm/xe/xe_pci.c#n338   ... still requires force_probe as of linux-next 20240816

**Review by ranj063 — APPROVED** [review]

**Comment by marc-hb** [comment]
It's Friday night now but Monday 18th would be a great time to merge this because the test results are stable and generally good except for the two known alsabat and suspend/resume issues on LNL.

**Review by fredoh9 — APPROVED** [review]
I'm able to test legacy HDA on LNL with this PR.
Without this, legacy HDA probe didn't work.

**Comment by ujfalusi** [comment]
@kv2019i, so, we declare xe stable even if the display guys are not comfortable with it? ;)

**Comment by kv2019i** [comment]
@ujfalusi wrote:
> @kv2019i, so, we declare xe stable even if the display guys are not comfortable with it? ;)

We are just enabling testing. I'm not proposing to send this upstream after all.

**Comment by marc-hb** [comment]
Good news: the sky didn't fall. Same LNL failures as usual into today's daily run 44893

A big caveat though: we disabled `gdm` a long time ago. Because it does not work, see below. Hopefully that does not matter for audio over HDMI/DP.
```
ug 19 23:52:51 jf-lnlm-rvp-nocodec-3 systemd[12296]: Starting GNOME Session Manager (session: ubuntu)...
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-session[12503]: gnome-session-binary[12503]: WARNING: Could not parse desktop file pulseaudio.desktop or it references a not found TryExec binary
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-session-binary[12503]: WARNING: Could not parse desktop file pulseaudio.desktop or it references a not found TryExec binary
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-keyring-daemon[12433]: The SSH agent was already initialized
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-keyring-ssh.desktop[12521]: SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-session[12503]: gnome-session-binary[12503]: GnomeDesktop-WARNING: Could not create transient scope for PID 12516: GDBus.Error:org.freedesktop.DBus.Error.UnixProcessIdUnknown: Process with ID 12516 does not exist.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-session-binary[12503]: GnomeDesktop-WARNING: Could not create transient scope for PID 12516: GDBus.Error:org.freedesktop.DBus.Error.UnixProcessIdUnknown: Process with ID 12516 does not exist.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-keyring-daemon[12433]: The Secret Service was already initialized
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-keyring-secrets.desktop[12522]: SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-keyring-daemon[12433]: The PKCS#11 component was already initialized
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-keyring-pkcs11.desktop[12523]: SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: app-gnome-gnome\x2dkeyring\x2dpkcs11-12520.scope: Couldn't move process 12520 to requested cgroup '/user.slice/user-1000.slice/user@1000.service/app.slice/app-gnome-gnome\x2dkeyring\x2dpkcs11-12520.scope': No such process
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: app-gnome-gnome\x2dkeyring\x2dpkcs11-12520.scope: Failed to add PIDs to scope's control group: No such process
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: app-gnome-gnome\x2dkeyring\x2dpkcs11-12520.scope: Failed with result 'resources'.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Failed to start Application launched by gnome-session-binary.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: app-gnome-gnome\x2dkeyring\x2dsecrets-12519.scope: Couldn't move process 12519 to requested cgroup '/user.slice/user-1000.slice/user@1000.service/app.slice/app-gnome-gnome\x2dkeyring\x2dsecrets-12519.scope': No such process
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: app-gnome-gnome\x2dkeyring\x2dsecrets-12519.scope: Failed to add PIDs to scope's control group: No such process
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: app-gnome-gnome\x2dkeyring\x2dsecrets-12519.scope: Failed with result 'resources'.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Failed to start Application launched by gnome-session-binary.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Started GNOME Session Manager (session: ubuntu).
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-session[12503]: gnome-session-binary[12503]: GnomeDesktop-WARNING: Could not create transient scope for PID 12524: GDBus.Error:org.freedesktop.DBus.Error.UnixProcessIdUnknown: Process with ID 12524 does not exist.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-session-binary[12503]: GnomeDesktop-WARNING: Could not create transient scope for PID 12524: GDBus.Error:org.freedesktop.DBus.Error.UnixProcessIdUnknown: Process with ID 12524 does not exist.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Started Application launched by gnome-session-binary.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Reached target GNOME Session Manager is ready.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Starting GNOME Shell on Wayland...
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Starting GNOME Shell on X11...
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: org.gnome.Shell@x11.service: Skipped due to 'exec-condition'.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Condition check resulted in GNOME Shell on X11 being skipped.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: org.gnome.Shell@x11.service: Scheduled restart job, restart counter is at 1.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Stopped GNOME Shell on X11.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Starting GNOME Shell on X11...
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: org.gnome.Shell@x11.service: Skipped due to 'exec-condition'.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Condition check resulted in GNOME Shell on X11 being skipped.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: org.gnome.Shell@x11.service: Scheduled restart job, restart counter is at 2.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Stopped GNOME Shell on X11.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Starting GNOME Shell on X11...
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: org.gnome.Shell@x11.service: Skipped due to 'exec-condition'.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Condition check resulted in GNOME Shell on X11 being skipped.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: org.gnome.Shell@x11.service: Scheduled restart job, restart counter is at 3.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Stopped GNOME Shell on X11.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: org.gnome.Shell@x11.service: Start request repeated too quickly.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: org.gnome.Shell@x11.service: Skipped due to 'exec-condition'.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 systemd[12296]: Started GNOME Shell on X11.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-shell[12534]: Running GNOME Shell (using mutter 42.9) as a Wayland display server
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-shell[12534]: Device '/dev/dri/card0' prefers shadow buffer
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 kernel: [drm] DisplayID checksum invalid, remainder is 248
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 kernel: [drm] DisplayID checksum invalid, remainder is 248
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 kernel: [drm] DisplayID checksum invalid, remainder is 248
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 kernel: [drm] DisplayID checksum invalid, remainder is 248
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 kernel: [drm] DisplayID checksum invalid, remainder is 248
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 kernel: [drm] DisplayID checksum invalid, remainder is 248
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 kernel: [drm] DisplayID checksum invalid, remainder is 248
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 kernel: [drm] DisplayID checksum invalid, remainder is 248
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-shell[12534]: Added device '/dev/dri/card0' (xe) using atomic mode setting.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-shell[12534]: MESA: warning: Driver does not support the 0x64a0 PCI ID.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-shell[12534]: MESA: warning: Driver does not support the 0x64a0 PCI ID.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-shell[12534]: Failed to initialize accelerated iGPU/dGPU framebuffer sharing: Not hardware accelerated
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-shell[12534]: MESA: warning: Driver does not support the 0x64a0 PCI ID.
Aug 19 23:52:52 jf-lnlm-rvp-nocodec-3 gnome-shell[12534]: libEGL warning: egl: failed to create dri2 screen
```

**Pull request merged**